### PR TITLE
Add BDD/Unit testing for nginx configuration.

### PR DESCRIPTION
**Pull request by steveworley**
- Adds testing framework as per govcmslagoon.
- Adds no-robots nginx conf.
- Updates CI to run tests.

### CODE REVIEW

File: .docker/images/nginx/tests/bootstrap.php [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+}
+
+function curl_get_content($path = "/", $opts = NULL)
+{
+  $uri = getenv('LOCALDEV_URL') ?: 'http://nginx:8080';
+
+  $response = null;
+  $path = '/' . ltrim($path, '/');
+
+  exec("docker-compose exec -T --env LAGOON_IMAGE_VERSION_PHP=7.1 test curl {$uri}{$path} {$opts} 2>/dev/null", $response);
```
Comment (srowlands) [inline .docker/images/nginx/tests/bootstrap.php]: GovCMS8 is PHP7.2 by default

Comment (srowlands) [reply .docker/images/nginx/tests/bootstrap.php]: Although.. actually what is this env var used for in this context? 

`LAGOON_IMAGE_VERSION_PHP` expects a value like `-v0.23.2`, (as per `.env.default`) - but I'm unclear why this is required?

File: docker-compose.yml [hunk 2]
```diff
@@ -22,7 +22,7 @@ services:
       args:
         GOVCMS_PROJECT_VERSION: ${GOVCMS_PROJECT_VERSION:-^1.0}
         DRUPAL_CORE_VERSION: ${DRUPAL_CORE_VERSION:-^8.7}
-        LAGOON_IMAGE_VERSION: ${LAGOON_IMAGE_VERSION_PHP}
+        LAGOON_IMAGE_VERSION: ${LAGOON_IMAGE_VERSION_PHP:-}
```
Comment (simesy) [inline docker-compose.yml]: Just checking what this change is since LAGOON_IMAGE_VERSION_PHP was removed from docker-compose.yml in master.

Comment (steveworley) [reply docker-compose.yml]: Yeah, I'll rebase with master before the merge. It was just to remove some warnings that were annoying me :joy:

**Review by stooit — CHANGES_REQUESTED** [review]
Rebased against latest `develop` to strip change to docker-compose

**Review by stooit — APPROVED** [review]

**Pull request merged**